### PR TITLE
ENH: CIFTI / fsLR density

### PR DESCRIPTION
**Pull request by mgxd**
- Adds new `density` output to `GenerateCifti` interface to allow tracking CIFTI file density (grayordinates)
- Adds `density` to `NameCifti` to differentiate low/high res data.
- Adds `template_kwargs` to `GiftiNameSource` to allow more flexible formatting

### CODE REVIEW

File: niworkflows/interfaces/surf.py [hunk 1]
```diff
@@ -131,6 +132,15 @@ class GiftiNameSource(SimpleInterface):
     >>> res.outputs.out_name
     'space-fsaverage.R.func'
 
+    >>> namer = GiftiNameSource()
+    >>> namer.inputs.pattern = r'(?P<LR>[lr])h.(?P<space>\w+).gii'
+    >>> namer.inputs.template = r'space-{space}.{den}.{LR}.func'
```
Comment (oesteban) [inline niworkflows/interfaces/surf.py]:
```suggestion
    >>> namer.inputs.template = r'space-{space}_den-{density}_hemi-{hemi}.func'
```

File: niworkflows/interfaces/surf.py [hunk 2]
```diff
@@ -131,6 +132,15 @@ class GiftiNameSource(SimpleInterface):
     >>> res.outputs.out_name
     'space-fsaverage.R.func'
 
+    >>> namer = GiftiNameSource()
+    >>> namer.inputs.pattern = r'(?P<LR>[lr])h.(?P<space>\w+).gii'
```
Comment (oesteban) [inline niworkflows/interfaces/surf.py]:
```suggestion
    >>> namer.inputs.pattern = r'(?P<hemi>[lr])h.(?P<space>\w+).gii'
```

File: niworkflows/interfaces/surf.py [hunk 3]
```diff
@@ -131,6 +132,15 @@ class GiftiNameSource(SimpleInterface):
...
+    >>> namer.inputs.template = r'space-{space}.{den}.{LR}.func'
+    >>> namer.inputs.in_file = 'rh.fsaverage.gii'
+    >>> namer.inputs.template_kwargs = {'den': '10k'}
```
Comment (oesteban) [inline niworkflows/interfaces/surf.py]:
```suggestion
    >>> namer.inputs.template_kwargs = {'density': '10k'}
```

File: niworkflows/interfaces/surf.py [hunk 4]
```diff
@@ -131,6 +132,15 @@ class GiftiNameSource(SimpleInterface):
...
+    >>> res = namer.run()
+    >>> res.outputs.out_name
+    'space-fsaverage.10k.R.func'
```
Comment (oesteban) [inline niworkflows/interfaces/surf.py]:
```suggestion
    'space-fsaverage_den-10k_hemi-R.func'
```

File: niworkflows/interfaces/surf.py [hunk 5]
```diff
@@ -147,6 +157,8 @@ def _run_interface(self, runtime):
         in_file = os.path.basename(self.inputs.in_file)
         info = in_format.match(in_file).groupdict()
         info['LR'] = info['LR'].upper()
```
Comment (oesteban) [inline niworkflows/interfaces/surf.py]:
```suggestion
         info['hemi'] = info['hemi'].upper()
```

File: niworkflows/interfaces/surf.py (hunk 2 above)
Comment (mgxd) [reply niworkflows/interfaces/surf.py]: i think this was meant to adhere to the GIFTI standard, not BIDS

File: niworkflows/interfaces/surf.py (hunk 1 above)
Comment (mgxd) [reply niworkflows/interfaces/surf.py]: this would require a fair bit of structuring on fmriprep's side - mainly hitting all fsaverage surfaces

Comment (oesteban) [reply niworkflows/interfaces/surf.py]: but you can redefine the input of the nipype interface right?

Comment (mgxd) [reply niworkflows/interfaces/surf.py]: oh, i thought this comment was on the fmriprep PR.

changing `den` to `density` sounds reasonable, but BIDS-ifying the whole thing seems unnecessary, at least for these changes

Comment (oesteban) [reply niworkflows/interfaces/surf.py]: Yeah - since this is going to show up in the documentation as an example, I'd go with a BIDS-y example.

**Comment by mgxd** [comment]
👍

File: niworkflows/interfaces/cifti.py [hunk 6]
```diff
@@ -119,24 +127,44 @@ class _CiftiNameSourceInputSpec(BaseInterfaceInputSpec):
...
 
 
 class CiftiNameSource(SimpleInterface):
-    """Construct new filename based on unique label of spaces used to generate a CIFTI file."""
+    """
+    Construct new filename based on unique label of spaces used to generate a CIFTI file.
+
+    Examples
+    --------
+
```
Comment (oesteban) [inline niworkflows/interfaces/cifti.py]: Please remove emtpy-line between header section and example

File: niworkflows/interfaces/surf.py (hunk 2 above)
Comment (oesteban) [reply niworkflows/interfaces/surf.py]: not sure what you mean here? Does GIFTI define an "LR" metadata?

still, space and density are bids-like... certainly unrelated to GIFTI.

I would apply this change also for readability - LR could be easily confused with the pattern (lr) and the template (fsLR).

**Comment by mgxd** [comment]
if there are no objections, I'll merge this in later today and cut a new release

**Pull request merged**